core/eth/state.go

```diff
@@ -38,7 +45,12 @@ func (cs *ChainState) GetOperatorState(ctx context.Context, blockNumber uint, qu
 		return nil, err
 	}
 
-	return getOperatorState(operatorsByQuorum, uint32(blockNumber))
+	socketMap, err := cs.buildSocketMap(ctx, operatorsByQuorum)
```

Comment:
does this mean whenever we call this method, it will query the chain `# operators` times? 
this seems very inefficient, should we cache this?

Reply:
yes I expect this too. I can add socket map to the ChainState cache and only query if the operator is missing from the socket map